iodata/formats/mwfn.py

```diff
@@ -0,0 +1,367 @@
+# D shell: XX, YY, ZZ, XY, XZ, YZ
+# F shell: XXX, YYY, ZZZ, XYY, XXY, XXZ, XZZ, YZZ, YYZ, XYZ
+# G shell: ZZZZ, YZZZ, YYZZ, YYYZ, YYYY, XZZZ, XYZZ, XYYZ, XYYY, XXZZ, XXYZ, XXYY, XXXZ, XXXY, XXXX
+# H shell: ZZZZZ, YZZZZ, YYZZZ, YYYZZ, YYYYZ, YYYYY, XZZZZ, XYZZZ, XYYZZ, XYYYZ, XYYYY, XXZZZ,
+#           XXYZZ, XXYYZ, XXYYY, XXXZZ, XXXYZ, XXXYY, XXXXZ, XXXXY, XXXXX
+# For pure shells, the order is
+# D shell: D 0, D+1, D-1, D+2, D-2
+# F shell: F 0, F+1, F-1, F+2, F-2, F+3, F-3
+# G shell: G 0, G+1, G-1, G+2, G-2, G+3, G-3, G+4, G-4
+
+CONVENTIONS = {
+    (9, 'p'): HORTON2_CONVENTIONS[(9, 'p')],
+    (8, 'p'): HORTON2_CONVENTIONS[(8, 'p')],
+    (7, 'p'): HORTON2_CONVENTIONS[(7, 'p')],
+    (6, 'p'): HORTON2_CONVENTIONS[(6, 'p')],
+    (5, 'p'): HORTON2_CONVENTIONS[(5, 'p')],
+    (4, 'p'): HORTON2_CONVENTIONS[(4, 'p')],
+    (3, 'p'): HORTON2_CONVENTIONS[(3, 'p')],
+    (2, 'p'): HORTON2_CONVENTIONS[(2, 'p')],
+    (0, 'c'): ['1'],
+    (1, 'c'): ['x', 'y', 'z'],
+    (2, 'c'): ['xx', 'yy', 'zz', 'xy', 'xz', 'yz'],
+    (3, 'c'): ['xxx', 'yyy', 'zzz', 'xyy', 'xxy', 'xxz', 'xzz', 'yzz', 'yyz', 'xyz'],
+    (4, 'c'): HORTON2_CONVENTIONS[(4, 'c')][::-1],
+    (5, 'c'): HORTON2_CONVENTIONS[(5, 'c')][::-1],
+    (6, 'c'): HORTON2_CONVENTIONS[(6, 'c')][::-1],
+    (7, 'c'): HORTON2_CONVENTIONS[(7, 'c')][::-1],
+    (8, 'c'): HORTON2_CONVENTIONS[(8, 'c')][::-1],
+    (9, 'c'): HORTON2_CONVENTIONS[(9, 'c')][::-1],
+}
+
+
+def _load_helper_opener(lit: LineIterator, keys: list) -> Tuple[int, float, float,
+                                                                float, float, float, int]:
+    """Read initial variables."""
+    max_count = len(keys)
+    count = 0
+    d = {}
+    while count < max_count:
+        line = next(lit)
+        for name in keys:
+            if name in line:
+                d[name] = line.split('=')[1].strip()
+                count += 1
+    return int(d['Wfntype']), float(d['Charge']), float(d['Naelec']), float(d['Nbelec']),\
+        float(d['E_tot']), float(d['VT_ratio']), int(d['Ncenter'])
+
+
+def _load_helper_basis(lit: LineIterator) -> Tuple[int, int, int, int, int]:
+    """Read initial variables."""
+    # Nprims must be last or else it gets read in with Nprimshell
+    basis_keywords = ["Nbasis", "Nindbasis", "Nshell", "Nprimshell", "Nprims", ]
+    max_count = len(basis_keywords)
+    count = 0
+    d = {}
+    next(lit)
+    while count < max_count:
+        line = next(lit)
+        for name in basis_keywords:
+            if name in line:
+                d[name] = int(line.split('=')[1].strip())
+                count += 1
+                break
+    return d['Nbasis'], d['Nindbasis'], d['Nprims'], d['Nshell'], d['Nprimshell']
+
+
+def _load_helper_atoms(lit: LineIterator, num_atoms: int) -> Tuple[np.ndarray, np.ndarray,
+                                                                   np.ndarray]:
+    """Read the coordinates of the atoms."""
+    atnums = np.empty(num_atoms, int)
```

Comment:
We typically use `natom` for this in IOData. (It is a naming convention from the fortran world, which may seem odd, but it works well for keeping names short.) `s` is not included to reserve plural for arrays, and for brevity of course.

Reply:
I also use natom in my fortran. this change makes sense to me.

Reply:
Actually, this is for atomic numbers, i.e., oxygen is 8, carbon is 6. I have changed the variable to `atomic_numbers` to provide a better description. Is this alright?

Reply:
I'm sorry for the confusion. What I meant is `num_atoms` -> `natom`.

It would be good to use the names from the ``IOData`` class, see https://github.com/theochem/iodata/blob/master/iodata/iodata.py#L64 (We've discussed these upfront some time ago and tried to find compromises between brevity and clarity.) In general, reusing names from the iodata modules like `iodata.iodata`, `iodata.api`, etc. results in consistent names, which makes the code easier to understand.